ros/lib/rbac_interface.py

```diff
@@ -0,0 +1,138 @@
+from urllib.parse import urljoin
+from http import HTTPStatus
+from flask_restful import abort
+from .config import RBAC_SVC_URL, PATH_PREFIX
+import requests
+
+
+RBAC_SVC_ENDPOINT = "/api/rbac/v1/access/?application=%s"
+AUTH_HEADER_NAME = "X-RH-IDENTITY"
+VALID_HTTP_VERBS = ["get", "options", "head", "post", "put", "patch", "delete"]
+
+
+def fetch_url(url, auth_header, logger, method="get"):
+    """
+    Helper to make a single request *** Add to this comment once functionality done.
+    """
+    if method not in VALID_HTTP_VERBS:
+        abort(
+            HTTPStatus.METHOD_NOT_ALLOWED, message="'%s' is not valid HTTP method." % method
+        )
+
+    logger.debug("Fetching %s" % url)
+
+    response = requests.request(method, url, headers=auth_header)
+    logger.debug("fetched %s" % url)
+    _validate_service_response(response, logger, auth_header)
+    return response.json()
+
+
+def _validate_service_response(response, logger, auth_header):
+    """
+    Raise an exception if the response was not what we expected.
+    """
+    print(response.status_code)
+    if response.status_code == requests.codes.not_found:
+        abort(
+            response.status_code, message="%s error received from service: %s" % response.text
+        )
+
+    if response.status_code in [requests.codes.forbidden, requests.codes.unauthorized]:
+        logger.info(
+            "%s error received from service: %s" % (response.status_code, response.text)
+        )
+        # Log identity header if 401 (unauthorized)
+        if response.status_code == requests.codes.unauthorized:
+            if isinstance(auth_header, dict) and AUTH_HEADER_NAME in auth_header:
+                logger.info("identity '%s'" % get_key_from_headers(auth_header))
+            else:
+                logger.info("no identity or no key")
+        abort(
+            response.status_code, message=response.text
+        )
+
+    if response.status_code != requests.codes.ok:
+        abort(
+            response.status_code, message="Error received from backend service"
+        )
```

Comment:
Will it be possible to keep consistency between message values? 
Definitely, `response.text` value will be different for each exception but on line +41, you have added substring `error received from service: `, line +51 without this substring and here it includes somewhat same substring.

Reply:
done.